pkg/controller/storageversiongc/gc_controller_test.go

```diff
@@ -39,8 +41,12 @@ func setupController(ctx context.Context, clientset kubernetes.Interface) {
 	storageVersionInformer := informerFactory.Internal().V1alpha1().StorageVersions()
 
 	controller := NewStorageVersionGC(ctx, clientset, leaseInformer, storageVersionInformer)
+	informerFactory.Start(ctx.Done())
+	// Using this ensure informer caches are fully populated before starting the controller.
+	if !cache.WaitForCacheSync(ctx.Done(), controller.leasesSynced, controller.storageVersionSynced) {
```

Comment:
Don't think this is strictly necessary. We wait for the sync in `controller.Run()` already. https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/storageversiongc/gc_controller.go#L103-L106

+1 to polling instead of checking condition once though.

Reply:
The PollWithContextTimeout addition should catch those flakes though? Overall I don't have a strong opinion so will defer to the approvers.